libupload.groovy

```diff
@@ -41,7 +41,10 @@ def upload_to_clouds(pipecfg, basearch, buildID, stream) {
             withCredentials(creds) {
                 utils.syncCredentialsIfInRemoteSession(["AWS_BUILD_UPLOAD_CONFIG"])
                 def c = pipecfg.clouds.aws
-                def grant_user_args = c.test_accounts?.collect{"--grant-user ${it}"}.join(" ")
+                def grant_user_args = ""
+                if (c.test_accounts) {
+                    grant_user_args = c.test_accounts.collect{"--grant-user ${it}"}.join(" ")
```

Comment:
Not super familiar with Groovy but shouldn't this be handled by null safe operator (?.) that we already used.
Or maybe adding an additional null safe operator 
`def grant_user_args = c.test_accounts?.collect{"--grant-user ${it}"}?.join(" ")`

Reply:
Good point. Thanks for raising this. The Groovy documentation is [not very thorough](http://groovy-lang.org/operators.html#_safe_navigation_operator) around this operator, but I think the reason why the previous code doesn't work is because `.` and `?.` have equal precedence: http://groovy-lang.org/operators.html#_operator_precedence. So it went left to right, converting `c.test_accounts?.collect{}` to `null`.

I think your suggestion would work, but the problem is that we don't want to render a literal `"null"` in the `cosa buildextend-aws` below. Interestingly, `null.collect{}` actually works fine (returns an empty list), so we don't even need the first `?`. A slight tweak on your suggestion then is:

```groovy
def grant_user_args = c.test_accounts.collect{"--grant-user ${it}"}?.join(" ")
```

I.e. relying on `null.collect{}` returning an empty list. But this is so subtle that it's probably worth the more verbose version.